### PR TITLE
sln reader/writer

### PR DESCRIPTION
**Pull request by jgoshi**
@piotrpMSFT @livarcocc @krwq @blackdwarf

### CODE REVIEW

**Comment by jgoshi** [comment]
test CentOS7.1 x64 Debug Build

**Comment by jgoshi** [comment]
test Debian8.2 x64 Debug Build

File: test/Microsoft.DotNet.Cli.Sln.Internal.Tests/Microsoft.DotNet.Cli.Sln.Internal.Tests.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+//using System.Linq;
```
Comment (krwq) [inline test/Microsoft.DotNet.Cli.Sln.Internal.Tests/Microsoft.DotNet.Cli.Sln.Internal.Tests.cs]: commented out code

File: src/Microsoft.DotNet.Cli.Sln.Internal/Microsoft.DotNet.Cli.Sln.Internal.csproj [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    </PackageReference>
+    <PackageReference Include="System.Text.RegularExpressions">
+      <Version>4.1.0</Version>
+    </PackageReference>
+    <PackageReference Include="NETStandard.Library">
+      <Version>1.6.0</Version>
+    </PackageReference>
+  </ItemGroup>
+
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Sln.Internal/Microsoft.DotNet.Cli.Sln.Internal.csproj]: I don't think we need this. We have this in migrated projects to keep parity with PJ projects that had it defined there. Given that this is a new project, I don't think we really need it.

Comment (jgoshi) [inline src/Microsoft.DotNet.Cli.Sln.Internal/Microsoft.DotNet.Cli.Sln.Internal.csproj]: Ok I will remove this PropertyGroup

**Comment by jgoshi** [comment]
@MattGertz Can I get approval for this one? @blackdwarf is working on getting license approval, but once we have that I'd like to merge this one. It's basically a library I need to start work on migrating sln files (so the sln file can reference the migrated csproj files properly).

**Comment by MattGertz** [comment]
Approved pending license approval.

**Comment by blackdwarf** [comment]
LGTM, license wise

**Pull request merged**